### PR TITLE
AMP integration with Index Exchange

### PR DESCRIPTION
**Pull request by indexexchange**
Added amp-ad tag of type "ix"

### CODE REVIEW

**Comment by indexexchange** [comment]
I signed it!

**Comment by IndexExchange-JerryLi** [comment]
Hey @lannka 
I'll be the contact person for this integration. 

I took a look at the CI results, there isn't any log outputs, do you happen to know what are the cases that failed?

Also I noted that you've been asking companies to implement renderStart and noContentAvailable. IndexExchange is an ad exchange, so we'll always return an ad for each request, so noContentAvailable will not be applicable to us. As for renderStart, we are planning on looking into it on the next update. It is not trivial at this moment to alert AMP when ads starts rendering. 

Please let me know if you got any other questions,

Sincerely,
Jerry

**Comment by IndexExchange-JerryLi** [comment]
Looks like the logs were posted late, and I saw the problem. It is due to merge that the alphabetical order were changed. I'll be making an update and submit it again

**Comment by ixpaciga** [comment]
Please note this should be covered under the corporate CLA signed with Index Exchange on Oct 24, 2016 11:04 PDT

File: ads/ix.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/ix.js]: 2016

File: ads/ix.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+import {writeScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function ix(global, data) {
```
Comment (zhouyx) [inline ads/ix.js]: I see that you request mandatory data attributes, and accept optional data attributes. I would recommend using the `validateData` here to check those attributes exist. 
https://github.com/ampproject/amphtml/blob/master/3p/3p.js#L215

File: examples/ads.amp.html [hunk 3]
```diff
@@ -110,6 +110,7 @@
     <a href="#gmossp" class="broken"></a> |
     <a href="#ibillboard">iBILLBOARD</a> |
     <a href="#imobile">I-Mobile</a> |
+    <a href="#ix">Index Exchange</a> |
```
Comment (zhouyx) [inline examples/ads.amp.html]: please follow Alphabetical order.

**Comment by zhouyx** [comment]
@indexexchange @IndexExchange-JerryLi @ixpaciga Looks good in general, few comments.

**Comment by IndexExchange-JerryLi** [comment]
Thanks @zhouyx, I'll make the updates and submit the pull request again.

**Comment by zhouyx** [comment]
@indexexchange Any update here?

**Comment by IndexExchange-JerryLi** [comment]
Sorry for the delay. We have pushed new changes. Let us know if everything is satisfactory.

**Comment by zhouyx** [comment]
also need to change `amp-ad.md` file
>>>
Also please follow https://github.com/ampproject/amphtml/blob/master/ads/README.md#files-to-change here to include change for /extensions/amp-ad/amp-ad.md

Another thing is we have a [renderStart API ](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis)that helps AMP to provide better ad loading experience. 
Could you take a look and implement it (renderStart + noContentAvailable)?
Let me know if there's any question.

**Comment by IndexExchange-JerryLi** [comment]
We have updated the `amp-ad.md file`.

For noContentAvailable, we always return content for each request, so it is not applicable to us. As for renderStart, we are planning on looking into it on the later update. It is not trivial at this moment to alert AMP when ads starts rendering.

**Comment by zhouyx** [comment]
LGTM. ping @lannka to take a final look.

**Comment by zhouyx** [comment]
Thanks @IndexExchange-JerryLi for the PR.  Merged

**Pull request merged**